src/lib/journey/callbacks/select-idp/select-idp.svelte

```diff
@@ -53,20 +57,39 @@
       value: option.provider,
       text: option.uiConfig.buttonDisplayName,
     }));
+
+    prefersDarkTheme = window.matchMedia('(prefers-color-scheme: dark)').matches;
   }
 </script>
 
-<Grid num={idps.length}>
-  {#each idps as idp}
-    <Button type="button" width="auto" onClick={() => setBtnValue(idp.value)}
-      ><AppleIcon
-        classes="tw_inline-block tw_fill-current tw_text-secondary-dark dark:tw_text-secondary-light"
-      />
-      <T key="signInWith" />
+{#each idps as idp}
+  <Grid num={1}>
+    <Button type="button" width="auto" onClick={() => setBtnValue(idp.value)}>
+      {#if idp.text.toUpperCase().includes('APPLE')}
+        {#if prefersDarkTheme}
+          <AppleIconLight
+            classes="tw_inline-block tw_fill-current tw_text-secondary-dark dark:tw_text-secondary-light"
+          />
```

Comment:
@cerebrl I'd like your opinion on this section, for a couple of reasons.

Firstly, we're currently treating all of the social IDP provider buttons effectively the same from a styling perspective. In terms of button styles within the widget project we don't have many choices - e.g. light or dark, outline or filled seem to be our options. There are guidelines from each provider on how the button should be styled, worded and placed depending on what is around it. I plan to adjust the styling of each button to meet these requirements - my approach will likely be to create classes specific to each provider and pass them in to the Button component here. If you have any thoughts on how else you'd like this to be done (or want to object to doing this at all), it would be good to discuss sooner rather than later.

Secondly, as I've implemented this here we're hard coding support for just these three IDPs, relying on the IDP name matching and making assumptions about the icon and styling to use. I think we should have an "else" option giving a generic icon and style if we don't recognise the IDP. I also think this could be made configurable, for example if someone wants to pass in configuration for a new IDP or a different logo. I'm just not sure that we should make it configurable, given these are the three supported providers by the SDK. Additionally if we style according to guidelines, I'm not sure if it is worth us making it super flexible as this may be something no one wants to do.

Anyway, please let me know what you think on the above points and we can discuss further. Thanks.

Reply:
Understood, I'm implementing something that will allow us to apply bgColor and textColor to buttons - also looking at additional properties that we might use if we ever decide to allow the uiConfig returned with the social callback to dictate style.

Reply:
As we discussed on slack, we are now styling the buttons using classes passed from the SelectIdp component. The result is that we can take advantage of the tailwind detection of light and dark mode.